aerleon/aclcheck_cmdline.py

```diff
@@ -17,72 +17,104 @@
+from argparse import ArgumentParser, RawTextHelpFormatter
 
 from aerleon.lib import aclcheck, naming, policy, yaml
+from aerleon.utils import config
 
 
 def main():
-    # TODO(robankeny): Lets move this to gflags
     usage = 'usage: %prog [options] arg'
-    _parser = OptionParser(usage)
-    _parser.add_option(
+    _parser = ArgumentParser(
+        prog='aclcheck',
+        formatter_class=RawTextHelpFormatter,
+    )
+    _parser.add_argument(
         '--definitions-directory',
-        dest='definitions',
+        dest='definitions_directory',
         help='definitions directory',
-        default='./def',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '--base-directory',
-        dest='basedir',
+        dest='base_directory',
         help='The base directory to look for include files.',
-        default='',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '-p',
         '--policy-file',
         dest='pol',
         help='policy file',
-        default='./policies/pol/sample_paloalto_yaml.yaml',
+        required=True,
+    )
+    _parser.add_argument(
+        '--config-file',
+        dest='config_file',
+        help='config file',
     )
-    _parser.add_option(
-        '-d', '--destination', dest='dst', help='destination IP', default='200.1.1.1'
+    _parser.add_argument('-d', '--destination', dest='dst', help='destination IP')
+    _parser.add_argument(
+        '-s',
+        '--source',
+        dest='src',
+        help='source IP',
     )
-    _parser.add_option('-s', '--source', dest='src', help='source IP', default='any')
-    _parser.add_option(
+    _parser.add_argument(
         '--proto',
         '--protocol',
         dest='proto',
         help='Protocol (tcp, udp, icmp, etc.)',
-        default='tcp',
-    )
-    _parser.add_option(
-        '--dport', '--destination-port', dest='dport', help='destination port', default='80'
     )
-    _parser.add_option(
-        '--sport', '--source-port', dest='sport', help='source port', default='1025'
-    )
-    (FLAGS, unused_args) = _parser.parse_args()
+    _parser.add_argument('--dport', '--destination-port', dest='dport', help='destination port')
+    _parser.add_argument('--sport', '--source-port', dest='sport', help='source port')
+    FLAGS = _parser.parse_args()
+
+    default_flags = {
+        'base_directory': './policies',
+        'definitions_directory': './def',
+        'pol': None,
+        'config_file': None,
+        'dst': '200.1.1.1',
+        'src': 'any',
+        'proto': 'any',
+        'dport': '80',
+        'sport': '1025',
+    }
 
-    defs = naming.Naming(FLAGS.definitions)
-    with open(FLAGS.pol) as f:
+    configs = {}
+    configs.update(default_flags)
+
+    if not (FLAGS.base_directory and FLAGS.definitions_directory):
+        common_flags = frozenset(['base_directory', 'definitions_directory'])
+        common_configs = config.load_config(config_file=FLAGS.config_file)
```

Comment:
Shouldn't you be handling the file not existing error here?

Reply:
No, see the docstring for config.load_config . FileNotFoundError is handled inside load_config as it is not an error if there is no aerleon.yml file (which will be the case for most users).

If opening the config file fails with IOError we do want to terminate the process and inform the user. In that case it might be a little better to catch and format the error.

https://peps.python.org/pep-3151/#common-errnos-with-ioerror

#### Common errnos with IOError

- EACCES: permission denied (for filesystem operations)
- EBADF: bad file descriptor (with select.epoll); read operation on a write-only GzipFile, or vice-versa
- EBUSY: device or resource busy
- EISDIR: is a directory (when trying to open())
- ENODEV: no such device
- ENOENT: no such file or directory (when trying to open())
- ETIMEDOUT: connection timed out

Reply:
I've updated the error handling behavior per slack convo.